MobileCenterCrashes/MobileCenterCrashes/MSCrashes.mm

```diff
@@ -770,25 +790,35 @@ - (NSMutableArray *)persistedCrashReports {
   NSMutableArray *persistedCrashReports = [NSMutableArray new];
 
   if ([self.crashesDir checkResourceIsReachableAndReturnError:&error]) {
-    NSArray *files =
-        [self.fileManager contentsOfDirectoryAtURL:self.crashesDir
+    NSArray *files = [self.fileManager contentsOfDirectoryAtURL:self.crashesDir
                         includingPropertiesForKeys:@[ NSURLNameKey, NSURLFileSizeKey, NSURLIsRegularFileKey ]
                                            options:NSDirectoryEnumerationOptions(0)
                                              error:&error];
+    if (!files) {
+      MSLogError([MSCrashes logTag], @"Couldn't get contents of directory \"%@\": %@", self.crashesDir, error.localizedDescription);
+      return persistedCrashReports;
+    }
     for (NSURL *fileURL in files) {
       NSString *fileName = nil;
-      [fileURL getResourceValue:&fileName forKey:NSURLNameKey error:&error];
+      if (![fileURL getResourceValue:&fileName forKey:NSURLNameKey error:&error]) {
+        MSLogWarning([MSCrashes logTag], @"Couldn't get resource value from \"%@\": %@", fileURL, error.localizedDescription);
+      }
       NSNumber *fileSizeNumber = nil;
-      [fileURL getResourceValue:&fileSizeNumber forKey:NSURLFileSizeKey error:&error];
+      if (![fileURL getResourceValue:&fileSizeNumber forKey:NSURLFileSizeKey error:&error]) {
+        MSLogWarning([MSCrashes logTag], @"Couldn't get resource value from \"%@\": %@", fileURL, error.localizedDescription);
+      }
       NSNumber *isRegular = nil;
-      [fileURL getResourceValue:&isRegular forKey:NSURLIsRegularFileKey error:&error];
-
+      if (![fileURL getResourceValue:&isRegular forKey:NSURLIsRegularFileKey error:&error]) {
+        MSLogWarning([MSCrashes logTag], @"Couldn't get resource value from \"%@\": %@", fileURL, error.localizedDescription);
+      }
       if ([isRegular boolValue] && [fileSizeNumber intValue] > 0 && ![fileName hasSuffix:@".DS_Store"] &&
           ![fileName hasSuffix:@".analyzer"] && ![fileName hasSuffix:@".plist"] && ![fileName hasSuffix:@".data"] &&
           ![fileName hasSuffix:@".meta"] && ![fileName hasSuffix:@".desc"]) {
         [persistedCrashReports addObject:fileURL];
       }
     }
+  } else if (error) {
+    MSLogWarning([MSCrashes logTag], @"Check \"%@\" error: %@", self.crashesDir, error.localizedDescription);
```

Comment:
I really doubt that we need this log, @clpolet @TroubleMakerBen. If yes, revise the message.